streams/src/test/java/org/apache/kafka/streams/processor/internals/ActiveTaskCreatorTest.java

```diff
@@ -67,40 +64,70 @@
     @Mock(type = MockType.NICE)
     private InternalTopologyBuilder builder;
     @Mock(type = MockType.NICE)
-    private StreamsConfig config;
-    @Mock(type = MockType.NICE)
     private StateDirectory stateDirectory;
     @Mock(type = MockType.NICE)
     private ChangelogReader changeLogReader;
-    @Mock(type = MockType.NICE)
-    private Consumer<byte[], byte[]> consumer;
-    @Mock(type = MockType.NICE)
-    private Admin adminClient;
 
     private final MockClientSupplier mockClientSupplier = new MockClientSupplier();
     final StreamsMetricsImpl streamsMetrics = new StreamsMetricsImpl(new Metrics(), "clientId", StreamsConfig.METRICS_LATEST);
+    final Map<String, Object> properties = mkMap(
+        mkEntry(StreamsConfig.APPLICATION_ID_CONFIG, "appId"),
+        mkEntry(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy:1234")
+    );
 
     private ActiveTaskCreator activeTaskCreator;
 
+
+
+    // non-EOS test
+
+    // functional test
+
     @Test
-    public void shouldFailForNonEosOnStreamsProducerPerTask() {
-        expect(config.getString(StreamsConfig.APPLICATION_ID_CONFIG)).andReturn("appId");
-        expect(config.getString(StreamsConfig.PROCESSING_GUARANTEE_CONFIG)).andReturn(StreamsConfig.AT_LEAST_ONCE);
-        expect(config.getProducerConfigs(anyString())).andReturn(Collections.emptyMap());
-        replay(config);
+    public void shouldCreateThreadProducerIfEosDisabled() {
```

Comment:
This test should no cover _all_ required tests for `ActiveTaskCreator` and thus addressed  on half of https://issues.apache.org/jira/browse/KAFKA-9676

Reply:
You mean `now`? :) If yes please feel free to resolve the ticket when you merge this.

Reply:
Yes, it should _now_ cover all required tests.

We still can't resolve KAFAK-9676 afterwards, because `StandbyTaskCreatorTest` is still lacking full test coverage.